airdrop/project/contracts/distributor-contract/src/main.sw

```diff
@@ -96,15 +96,15 @@ impl AirdropDistributor for Contract {
     ) {
         // If `end_block` is set to a value other than 0, we know that the contructor has already
         // been called.
-        require(storage.end_block == 0, InitError::AlreadyInitialized);
+        require(storage.end_block.read() == 0, InitError::AlreadyInitialized);
         require(msg_amount() > 0, InitError::CannotAirdropZeroTokens);
 
         let asset = msg_asset_id();
-        storage.end_block = height() + claim_time;
-        storage.merkle_root = Option::Some(merkle_root);
-        storage.asset = Option::Some(asset);
-        storage.number_of_leaves = number_of_leaves;
-        storage.admin = Option::Some(admin);
+        storage.end_block.write(height() + claim_time);
```

Comment:
That's a lot of writes. Would refactoring into a single object and writing that once be cheaper here @bitzoic ?

Reply:
If this is the case I'll perform the refactor in a separate issue